TLM/TLM/Patch/_VehicleAI/StartPathFindCommons.cs

```diff
@@ -0,0 +1,45 @@
+namespace TrafficManager.Patch._VehicleAI {
+    using System.Reflection;
+    using TrafficManager.Util;
+    using UnityEngine;
+    using HarmonyLib;
+    using System.Collections.Generic;
+    using System.Reflection.Emit;
+    using CSUtil.Commons;
+
+    public static class StartPathFindCommons {
+        // protected override bool StartPathFind(ushort vehicleID, ref Vehicle vehicleData, Vector3 startPos, Vector3 endPos, bool startBothWays, bool endBothWays, bool undergroundTarget)        delegate bool TargetDelegate(out uint unit, ref Randomizer randomizer, uint buildIndex,
+        delegate bool TargetDelegate(ushort vehicleID, ref Vehicle vehicleData, Vector3 startPos, Vector3 endPos, bool startBothWays, bool endBothWays, bool undergroundTarget);
+        public static MethodBase TargetMethod<T>() {
+            return TranspilerUtil.DeclaredMethod<TargetDelegate>(typeof(T), "StartPathFind");
+        }
+
+        /// <summary>
+        /// replaces max pos of 4800 with 8000 to support 81 tiles.
+        /// </summary>
+        public static IEnumerable<CodeInstruction> ReplaceMaxPosTranspiler(IEnumerable<CodeInstruction> instructions) {
+            // TODO [issue #] Should this be done in TMPE?
+            // see https://github.com/CitiesSkylinesMods/TMPE/issues/895#issuecomment-643111138
+            const float vanilaMaxPos = 4800f; // vanilla 25 tiles compatible value
+            const float newMaxPos = 8000f; // 81 tiles compatible value.
+
+            int counter = 0;
+            foreach (var instruction in instructions) {
+                bool is_ldfld_minCornerOffset =
+                    instruction.opcode == OpCodes.Ldc_R4 && ((float)instruction.operand) == vanilaMaxPos;
+                if (is_ldfld_minCornerOffset) {
+                    counter++;
+                    yield return new CodeInstruction(OpCodes.Ldc_R4, operand: newMaxPos);
+                } else {
+                    yield return instruction;
+                }
+            }
+
+            // if another mod has already made such replacement then this assertion fails and we would know :)
+            Shortcuts.Assert(counter > 0, "n>0");
```

Comment:
Seeing `Assertion failed: n>0` in a user log file isn't exactly helpful (remember the subsequent `Log._Debug()` is filtered from release builds).

Would it be more useful to user with something like this:

```cs
Log.WarningIf(counter == 0, () =>
    "Mod '81 Tiles': Compatibility patch failed in StartPathFindCommons.ReplaceMaxPosTranspiler()");
```

Reply:
No, this is not a sign of failure. 
This is a sign that there is another mod that has already fixed the 81 tile issue and therefore the existence of this transpiler in TMPE is redundant.
This will be preceded by the message:" replaced 0 instances ...." so it would be clear what 0 here is. Not to mention the stack trace.
This will not cause any functional problem for the user.

Reply:
> No, this is not a sign of failure.

Then why are we using `Assert()` will does a `Log.Error()` with message `Assertion failed: n>0` which user won't comprehend.

A `Log.Warning()` seems to be far more logical here. The message can change: `Unable to patch 'maxPos' (may cause pathfinding issues if the value is too low).`

`Log.Warning()` also outputs a stack trace.